src/Services/AppointmentService.php

```diff
@@ -199,4 +242,125 @@ protected function createResultRecordFromDatabaseResult($row)
     {
         return parent::createResultRecordFromDatabaseResult($row); // TODO: Change the autogenerated stub
     }
+
+    /**
+     * Returns a list of categories
+     * @return array
+     */
+    public function getCalendarCategories()
+    {
+        $sql = "SELECT pc_catid, pc_constant_id, pc_catname, pc_cattype,aco_spec FROM openemr_postcalendar_categories "
+        . " WHERE pc_active = 1 ORDER BY pc_seq";
+        return QueryUtils::fetchRecords($sql);
+    }
+
+    /**
+     * check to see if a status code exist as a check in
+     * @param $option
+     * @return bool
+     */
+    public function isCheckInStatus($option)
+    {
+        $row = sqlQuery("SELECT toggle_setting_1 FROM list_options WHERE " .
+            "list_id = 'apptstat' AND option_id = ? AND activity = 1", array($option));
+        if (empty($row['toggle_setting_1'])) {
+            return(false);
+        }
+
+        return(true);
+    }
+
+    /**
+     * check to see if a status code exist as a check out
+     * @param $option
+     * @return bool
+     */
+    public function isCheckOutStatus($option)
+    {
+        $row = sqlQuery("SELECT toggle_setting_2 FROM list_options WHERE " .
+            "list_id = 'apptstat' AND option_id = ? AND activity = 1", array($option));
+        if (empty($row['toggle_setting_2'])) {
+            return(false);
+        }
+
+        return(true);
+    }
+
+    /**
+     * Updates the status for an appointment.  TODO: should be refactored at some point to update the entire record
+     * @param $eid The id of the appointment event
+     * @param $status The status the appointment event should be set to.
+     */
+    public function updateAppointmentStatus($eid, $status)
```

Comment:
Check out the manage_tracker_status() function. That mechanism ensure that status changes are logged which is used in reporting mechanism (and any other data analytics a practice may want to do).